HPCC-28089 Report ‘No logging stack available’ in WUInfo response

wangkx opened the pull request:
For ECL WU on cloud environment, a ComponentLog weblink is shown
the Helpers page. It will not work if WsWorkunits cannot access
logging stack. WsWorkunits.WUInfo should check whether a logging
stack is available or not and report to ECLWatch.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

GordonSmith commented on esp/scm/ws_workunits_struct.ecm:
```diff
@@ -244,6 +244,7 @@ ESPStruct [nil_remove] ECLHelpFile
     [min_ver("1.44")] unsigned PID(0);
     [min_ver("1.58")] unsigned minActivityId(0);
     [min_ver("1.58")] unsigned maxActivityId(0);
+    [min_ver("1.94")] bool IsAvailable(true);
```
This really belongs in WsTopology as we display the logs in various places, not just for workunits...

wangkx replied:
This is for the ComponentLog (may be used for other ECLHelpFile in future if needed). The ComponentLog function is only for ECL Workunits in cloud environment. For now, the log function in WsTopology is for bare metal only (log files are always available). I think we should add this (and check this) if we add the logging stack support in WsTopology.

GordonSmith replied:
FYI The errors you see in the ECL Watch WU Details page are coming from the global logging query.

wangkx replied:
For the logs retrieved by ws_logaccess.GetLogs, could we use the existing ws_logaccess.GetLogAccessInfo to decide whether the log stack is available or not? Or add another flag (IsAvailable?) into ws_logaccess.GetLogAccessInfo?

wangkx committed:
HPCC-28089 Report ‘No logging stack available’ in WUInfo response
For ECL WU on cloud environment, a ComponentLog weblink is shown
the Helpers page. It will not work if WsWorkunits cannot access
logging stack. WsWorkunits.WUInfo should check whether a logging
stack is available or not and report to ECLWatch.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx commented:
Rebased to the latest 8.8.x.
